src/test/java/ikysil/training/dao/InMemoryOrderDaoTest.java

```diff
@@ -0,0 +1,73 @@
+package ikysil.training.dao;
+
+import ikysil.training.dao.impl.InMemoryOrderDao;
+import ikysil.training.ws.api.v1.dto.OrderDto;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class InMemoryOrderDaoTest {
+
+    private OrderDao dao = new InMemoryOrderDao();
+
+    private OrderDto order1 = new OrderDto("1144", "twenty one", "none", "3129", "firstuser@gmail.com");
+    private OrderDto order2 = new OrderDto("3257", "twenty two", "none", "4024", "seconduser@gmail.com");
+
+    @Before
+    public void saveOrderTest(){
+        dao.saveOrder(order1);
+        dao.saveOrder(order2);
+    }
+
+    @Test
+    public void findOrderTest(){
+        OrderDto testOrder1 = dao.findOrder("1144");
+        assertEquals(order1, testOrder1);
+
+        OrderDto testOrder2 = dao.findOrder("3257");
+        assertEquals(order2, testOrder2);
+    }
+
+    @Test
+    public void updateOrderTest() {
+        OrderDto newOrder = new OrderDto("3257","thirty three", "noneone", "2440", "newuser@gmail.com");
+        dao.updateOrder("3257", newOrder);
+        assertEquals(dao.findOrder("3257"), newOrder);
+    }
+
+    @Test
+    public void findAllTest(){
+        List<OrderDto> orders = dao.findAll();
+        assertEquals(2, orders.size());
+        deleteOrderTest();
+        orders = dao.findAll();
+        assertTrue(orders.isEmpty());
+        saveOrderTest();
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void insertNullTest(){
+        dao.saveOrder(null);
+    }
+
+    @Test
+    public void dublicatesTest(){
+        dao.saveOrder(order1);
+        dao.saveOrder(order1);
+        dao.saveOrder(order2);
+        dao.saveOrder(order2);
+    }
+
+    @After
+    public void deleteOrderTest(){
```

Comment:
weird name for @After method
I think you do not need @After here, as well better name may be `removeOrders`